portable/GCC/ARM_AARCH64/port.c

```diff
@@ -244,23 +248,47 @@ StackType_t * pxPortInitialiseStack( StackType_t * pxTopOfStack,
     *pxTopOfStack = ( StackType_t ) 0x00;         /* XZR - has no effect, used so there are an even number of registers. */
     pxTopOfStack--;
     *pxTopOfStack = ( StackType_t ) 0x00;         /* R30 - procedure call link register. */
-    pxTopOfStack--;
 
-    *pxTopOfStack = portINITIAL_PSTATE;
-    pxTopOfStack--;
-
-    *pxTopOfStack = ( StackType_t ) pxCode; /* Exception return address. */
     pxTopOfStack--;
+    *pxTopOfStack = portINITIAL_PSTATE;
 
-    /* The task will start with a critical nesting count of 0 as interrupts are
-     * enabled. */
-    *pxTopOfStack = portNO_CRITICAL_NESTING;
     pxTopOfStack--;
+    *pxTopOfStack = ( StackType_t ) pxCode; /* Exception return address. */
 
-    /* The task will start without a floating point context.  A task that uses
-     * the floating point hardware must call vPortTaskUsesFPU() before executing
-     * any floating point instructions. */
-    *pxTopOfStack = portNO_FLOATING_POINT_CONTEXT;
+    #if ( configUSE_TASK_FPU_SUPPORT == 1 )
```

Comment:
Should we update `portTASK_USES_FLOATING_POINT` in portmacro.h like this - https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/portable/GCC/ARM_CA9/portmacro.h#L125-L138?

And guard [vPortTaskUsesFPU ](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/portable/GCC/ARM_AARCH64/port.c#L443) with `#if ( configUSE_TASK_FPU_SUPPORT == 1 )`?

Reply:
Thanks for pointing out and adding.